src/cgr_gwas_qc/workflow/modules/thousand_genomes.smk

```diff
@@ -28,6 +28,31 @@ rule pull_b_allele_freq_from_1kg:
         "../scripts/bpm2abf.py"
 
 
+rule pull_b_allele_freq_from_1kg_bcfinput:
+    """Pulls the population level allele frequencies from the 1KG project.
+
+    ``verifyIDintensity`` requires population level allele frequencies
+    for its model. Here we use a custom script to pull out the allele B
+    frequencies (ABF) from the 1000 genomes project (1KG). To do this we
+    take each marker from the manifest file (BPM) and pull out ABF in the
+    1KG ``.vcf`` from the ``INFO`` column. The script allows pulling out
+    allele frequencies for different super populations but defaults to
+    ``AF`` which ignores super population.
+    """
+    input:
+        vcf_file=cfg.config.user_files.bcf,
```

Comment:
should this variable name be `bcf_file`?

Reply:
Renamed in 2d7964037ccb890f4fd39c7093444455d8126646 and 5009ca51c62d23cb865d0596f697f8456999762e

Reply:
Hi @jaamarks ,

Thanks again for taking out the time to review this PR. Appreciate your review in detail and checking for any issues. Overall, I think updated branch should reflect all/most changes requested. If anything else needed to be fixed, please feel free to let me know. It would be great if we have this merged into the next release too if possible. Might save sometime in realigning and resolving conflicts with new release. However, if you see any concerns in merging, we can hold on and do it later too.

Here are summary of the changes:

1. Rebased the branch as suggested.
2. Renamed variable to bcf_file wherever possible. Actually, the idea is that all functionality should be supported with vcf or bcf input seemlessly. And for the most part it should. However, I haven't done extensive testing with vcf_file. And atleast the plink import requires an explict --vcf or --bcf switch. For now, I have followed your suggestion to rename to bcf_file for clarity. Later on, we can test if vcf for all the worklows and if it works then we can revert the variable name to vcf_file for simplicity.
3. As discussed in the meeting, separating intensity subworkflow from contamination appeared more logical to me since the intensity checks specifically checks for intensity and not contamination. There is not much common between the two. Separating them also allows for testing and optimizing them independently if needed.

Feel free to let me know anything else needed. Thank you very much.